itou/www/signup/forms.py

```diff
@@ -53,6 +54,13 @@ def __init__(self, *args, **kwargs):
         self.organization = None
         self.new_organization = None
 
+    def clean_email(self):
+        # Must check if already exists
+        email = self.cleaned_data["email"]
+        if User.email_already_exists(email):
```

Comment:
Merci j'apprends des trucs :

```
    @classmethod
    def email_already_exists(cls, email):
        """
        RFC 5321 Part 2.4 states that only the domain portion of an email
        is case-insensitive. Consider toto@toto.com and TOTO@toto.com as
        the same email.
        """
        return cls.objects.filter(email__iexact=email).exists()
```

Reply:
En général on évite d'utiliser la classe `User` directement, on lui préfère la méthode `get_user_model` qui retourne le modèle d'utilisateur utilisé par le système d'authentification.

[Plus d'infos ici](https://docs.djangoproject.com/fr/3.0/topics/auth/customizing/#referencing-the-user-model).

Reply:
J'ignorais que l'on pouvais configurer l'auth pour changer de classe, merci du lien.
Par contre, ce n'est pas directement pas la classe `User` de `django.auth` qu'on utilise, mais une custom qui étends `AbstractUser`.